chain/chain/src/store.rs

```diff
@@ -2528,18 +2527,11 @@ mod tests {
         assert!(chain.clear_old_data().is_ok());
 
         assert!(chain.get_block(&blocks[0].hash()).is_ok());
+
+        // epoch didn't change so no data is garbage collected.
```

Comment:
Seems like this test is not testing anything now?

The correct fix of the test is probably to introduce an epoch switch.

Reply:
It tests that when epoch doesn't switch the blocks are not garbage collected. I had a separate test `test_garbage_collection` that tests the scenario you mentioned.

Reply:
@bowenwang1996 could you add epoch switching explicitly and add a bool parameter which says should epoch switching happen or not? Then you can remove separate test `test_garbage_collection`.

Reply:
I cannot set the epoch length in this test easily and also this test uses `KeyValueRuntime`. I prefer to have a separate test that uses the real runtime.

Reply:
Got it.
Does it mean that this change may affect GC fork test infrastructure negatively (gc_fork_common) in https://github.com/nearprotocol/nearcore/pull/2084? `gc_fork_common` uses `KeyValueRuntime` without real epoch switching.

Reply:
Yes. I think we need some modification to `get_gc_head_height` in `KeyValueRuntime`.

Reply:
Okay, please do.

Reply:
Also please rename the test so that it's more clear what it's testing